### PR TITLE
Update topo generator and add the topology for 96 downlinks, 32 uplinks and 2 peer links.

### PR DESCRIPTION
**Pull request by r12f**
### Description of PR

Summary:

This PR updates the topology generator python script to support T0 topology generation with specific downlinks, uplinks and peer links.

The script provides examples in its help:

```text
 ./generate_topo.py --help
Usage: generate_topo.py [OPTIONS]

  Generate a topology file for a device:

  Examples (in the ansible directory):
  ...
  - ./generate_topo.py -r t0 -k isolated -t t0-isolated -c 130 -p 128,129 -u 25,26,27,28,29,30,31,32
```

### Type of change

- [ ] Bug fix
- [x] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 202012
- [ ] 202205
- [ ] 202305
- [ ] 202311
- [ ] 202405

#### What is the motivation for this PR?

Creating topology can be tedious when the number of ports becomes large. Setting hundreds of VM configurations manually are not efficient.

#### How did you do it?

This PR updates the topology generator python script to support T0 topology generation with specific downlinks, uplinks and peer links.

Besides, it also creates an example topology t0-isolated-d96u32s2.yml, so we can unblock the device testing with 96 downlinks, 32 uplinks and 2 peer links.

#### Any platform specific information?

N/A

### CODE REVIEW

**Review by sdszhang — APPROVED** [review]
LGTM

File: ansible/templates/topo_t0-isolated.j2 [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+          - {{vm.dut_intf_ipv4}}
+          - {{vm.dut_intf_ipv6}}
+    interfaces:
+      Loopback0:
+        ipv4: {{vm.loopback_ipv4}}/32
+        ipv6: {{vm.loopback_ipv6}}/128
+      Ethernet1:
+        ipv4: {{vm.pc_intf_ipv4}}/31
+        ipv6: {{vm.pc_intf_ipv6}}/126
+    bp_interfaces:
```
Comment (ccroy-arista) [inline ansible/templates/topo_t0-isolated.j2:67]: Hi @r12f, I believe this should be `bp_interface`, likewise other references to bp_interfaces would need to be fixed as well (in this PR but also any other similar recent PRs).

Comment (r12f) [reply ansible/templates/topo_t0-isolated.j2:67]: just to double confirm, this is what I am seeing in existing topologies - t1-64. Does this mean it is also bad?

image

Comment (r12f) [reply ansible/templates/topo_t0-isolated.j2]: thanks Chris! done!

File: ansible/vars/topo_t0-isolated-d96u32s2.yml [hunk 2]
```diff
@@ -0,0 +1,948 @@
...
+    bgp:
+      asn: 64600
+      peers:
+        65100:
+          - 10.0.0.42
+          - fc00::55
+    interfaces:
+      Loopback0:
+        ipv4: 100.1.0.22/32
+        ipv6: 2064:100::16/128
```
Comment (r12f) [inline ansible/vars/topo_t0-isolated-d96u32s2.yml:714]: should use /64 for ipv6

Comment (r12f) [inline ansible/vars/topo_t0-isolated-d96u32s2.yml:714]: step need to be changed to /64.

File: ansible/vars/topo_t1-isolated-d224u8.yml [hunk 3]
```diff
@@ -2311,7 +2311,7 @@ configuration:
       Ethernet1:
         ipv4: 10.0.0.143/31
         ipv6: fc00::11e/126
-    bp_interfaces:
+    bp_interface:
       ipv4: 10.10.246.73/24
       ipv6: fc0a::49/64
   ARISTA69T0:
```
Comment (r12f) [inline ansible/vars/topo_t1-isolated-d224u8.yml:2317]: property is missing tor.

**Review by Blueve — APPROVED** [review]

**Review by bingwang-ms — APPROVED** [review]

**Pull request merged**